kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/SecureConfigInterpolator.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Interpolates a KafkaProxyFilter's configTemplate using a set of {@link SecureConfigProvider}s, producing a complete
+ * configuration for a filter, plus any necessary volumes and volumeMounts for files referred to from that configuration.
+ */
+public class SecureConfigInterpolator {
+
+    private static final Pattern PATTERN = Pattern.compile("^\\$\\{"
+            + "(?<providerName>[a-z]+)"
+            + ":(?<path>[a-zA-Z0-9_.-]+)"
+            + ":(?<key>[a-zA-Z0-9_.-]+)"
```

Comment:
Did you think about allowing directories to passed to the operand?  If we made the key part optional, the operand could receive a path reference to the directory, rather than a file within the the secret/configmap.

We could always add this later, if we see demand.   BTW.. this wouldn't work with [all-in-one](https://github.com/kubernetes/design-proposals-archive/blob/main/node/all-in-one-volume.md).